src/library/tactic/tactic_log.h

```diff
@@ -87,6 +88,7 @@ class tactic_log {
 
 private:
     mutable std::vector<tactic_invocation> m_invocs;
+    mutable std::vector<std::string> m_tspps; // tactic state pretty-printeds
     using state_map = std::unordered_map<summary, tactic_state_id, summary_hash, summary_eq>;
     mutable state_map m_state_map;
     mutable std::vector<summary> m_states;
```

Comment:
```suggestion
    mutable std::vector<std::pair<summary, std::string>> m_states;
```
If you push the pp into the state array then you only ever pretty print a state once. Correspondingly, the json export should put the pretty printed state at `root.states[i].pp` instead of `root.tspps`.

Reply:
Thanks. I just added it to the summary object instead.